src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3Driver.Linux.cs

```diff
@@ -313,6 +313,14 @@ protected internal override void Write(int pinNumber, PinValue value)
             *registerPointer = register;
         }
 
+        public void GetRegisters(out ulong* setRegister, out ulong* clearRegister)
```

Comment:
hmm, wondering if there is perhaps way to not expose pointers i.e.:

```csharp
public ulong SetRegister
{
  get { *_registerViewPointer->GPSET; }
  set {  *_registerViewPointer->GPSET = value; }
}

public ulong ClearRegister
{
  get { *_registerViewPointer->GPCLR; }
  set {  *_registerViewPointer->GPCLR = value; }
}
```

+ perhaps call to `Initialize()` (or just move it to constructor)

Reply:
> I think it was from @shaggygi
I don't recall anything related, but probably slept since then :smile: 
The only thing I can remember when it comes to registers is the reading output pin issue (#183)

Reply:
@krwq the only concern is we used to call Initialize() before when opening the Pin. I am not sure if we call Initialize() in the constructors will have any side effect. There should be a good reason we delay the initialization but I don't know what is it. if you are sure there is no any side effect, I can go with your idea.

Reply:
The other idea is to not do the initialization at all and anyone calling these registers properties has to open the pins (and that will do the initialization)

Reply:
@joperezr if you happen to read this - got any clues about the `Initialize` being delayed? cc: @buyaa-n